src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -44,10 +57,42 @@ const ApplicationPageWrapper: React.FC = () => {
         push,
         replace,
       })
+
+      if (sectionCode && page)
+        setCurrentSection(templateSections.find(({ code }) => code === sectionCode))
+    }
+  }, [isApplicationLoaded, sectionCode, page])
+
+  useEffect(() => {
+    const progressStructure = buildProgressInApplication({
```

Comment:
`comment`, with multiple effects, it would be good to name them somehow, a quick comment with their intended function would be great.

Reply:
No problem. I'll do that now